test/mp/test/test.cpp

```diff
@@ -49,12 +49,12 @@ namespace test {
 class TestSetup
 {
 public:
-    std::thread thread;
     std::function<void()> server_disconnect;
     std::function<void()> client_disconnect;
     std::promise<std::unique_ptr<ProxyClient<messages::FooInterface>>> client_promise;
     std::unique_ptr<ProxyClient<messages::FooInterface>> client;
     ProxyServer<messages::FooInterface>* server{nullptr};
+    std::thread thread;
```

Comment:
5975fef52c8f6092979acfaeac0c665d45436e65: maybe add a comment that this has to go after any member variable touched.